### PR TITLE
Separate Source Map and UglifyJS library file.

### PR DESCRIPTION
**Pull request by junaruga**
Hi,

Right now  `lib/uglify.js` is generated from `vendor/source-map/dist/source-map.js`, `./vendor/uglifyjs/bin/uglifyjs --self`, and a JavaScript code `MOZ_SourceMap = sourceMap;` in "rake js".
And all the JavaScript code is loaded in `lib/uglifier.rb#uglifyjs_source`.

I want to change this logic a little bit.
I want to make
`lib/source-map.js` from `vendor/source-map/dist/source-map.js`,
`lib/uglify.js` from `./vendor/uglifyjs/bin/uglifyjs --self` separately.

Because I am managing your gem package's RPM package in Fedora Project.
And I want to use system installed Source Map and UglifyJS for the RPM package uglifier, instead of using bundled these JavaScript files.
.

Is it possible?
I suppose this update is helpful for not only me, but also other Linux distribution's maintainers such as Ubuntu.

I could pass the tests by myself.

```
bundle exec rake js
bundle exec rake
```

Thanks.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -19,9 +19,10 @@ task :js do
 
   source = ""
   source << File.read("vendor/source-map/dist/source-map.js")
-  source << "MOZ_SourceMap = sourceMap;"
-  source << `./vendor/uglifyjs/bin/uglifyjs --self --comments /Copyright/`
+  File.write("lib/source-map.js", source)
 
+  source = ""
+  source << `./vendor/uglifyjs/bin/uglifyjs --self --comments /Copyright/`
```
Comment (lautis) [inline Rakefile]: Isn't concatenation redundant here? The output is written to a file as is.

File: lib/load-source-map.js [hunk 2]
```diff
@@ -0,0 +1 @@
+MOZ_SourceMap = sourceMap;
```
Comment (lautis) [inline lib/load-source-map.js]: Would it work if this was in uglifer.js?

File: Rakefile (hunk 1 above)
Comment (junaruga) [inline Rakefile]: Okay I will use FileUtils.cp and etc to shorten it.

File: lib/load-source-map.js (hunk 2 above)
Comment (junaruga) [inline lib/load-source-map.js]: That's good idea. It works. Let me update it.

**Comment by junaruga** [comment]
Hi, I pushed additional commit to my pull-request branch.
If you like it, you can merge with squash.
Otherwise, please tell me. I will modify it again.
Thanks.

**Comment by lautis** [comment]
Thank you! I'll try to remember to make a release tonight.

**Comment by junaruga** [comment]
@lautis okay, thank you!

**Pull request merged**